nautobot/utilities/views.py

```diff
@@ -151,3 +186,1062 @@ def get_return_url(self, request, obj=None):
 
         # If all else fails, return home. Ideally this should never happen.
         return reverse("home")
+
+
+class NautobotViewSetMixin:
+
+    routes = []
+    model = None
+    queryset = None
+    table = None
+    template_name = None
+    filterset = None
+    filterset_form = None
+    import_form = None
+    prefetch_related = []
+
+    def __init__(self):
+        self.routes = self.define_routes()
+
+    @property
+    def detail_queryset(self):
+        return self.model.objects.all()
+
+    @property
+    def table_queryset(self):
+        return self.model.objects.prefetch_related(*self.prefetch_related)
+
+    def define_routes(self):
+        return []
+
+    def get_extra_context(self, request, view_type, instance=None):
+        """
+        Return any additional context data for the template.
+        request: The current request
+        instance: The object being viewed
+        """
+        return {}
+
+    def get_template_name(self, view_type):
+        # Use "<app>/<model>_<view_type> if available, else fall back to generic templates
+        model_opts = self.model._meta
+        app_label = model_opts.app_label
+        if view_type == "detail":
+            return f"{app_label}/{model_opts.model_name}.html"
+
+        try:
+            select_template([f"{app_label}/{model_opts.model_name}_{view_type}.html"])
+            return f"{app_label}/{model_opts.model_name}_{view_type}.html"
+        except TemplateDoesNotExist:
+            return f"generic/object_{view_type}.html"
+
+
+class ObjectDetailViewMixin(NautobotViewSetMixin):
+    """
+    Retrieve a single object for display.
+    queryset: The base queryset for retrieving the object
+    template_name: Name of the template to use
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.routes = self.define_routes()
+
+    def define_routes(self):
+        return super().define_routes() + [
+            Route(
+                url=r"^{prefix}/{lookup}/$",
+                mapping={
+                    "get": "handle_object_detail_get",
+                },
+                name="{basename}",
+                detail=True,
+                initkwargs={"suffix": "Detail"},
+            ),
+        ]
+
+    def get_changelog_url(self, instance):
+        """Return the changelog URL for a given instance."""
+        meta = self.queryset.model._meta
+
+        # Don't try to generate a changelog_url for an ObjectChange.
+        if meta.model_name == "objectchange":
+            return None
```

Comment:
Perhaps instead, something like:

```suggestion
        if not issubclass(self.queryset.model, ChangeLoggedModel):
            return None
```